src/Cryptol/Eval/Generic.hs

```diff
@@ -608,6 +618,8 @@ zeroV sym ty = case ty of
   TVIntMod _ ->
     VInteger <$> integerLit sym 0
 
+  TVArray{} -> evalPanic "zeroV" ["Array not in class Zero"]
```

Comment:
Eventually it would probably good to add an instance `Zero (Array a b)` whenever we have `Zero b`, which could be implemented with `arrayConstant zero`. But we can skip it for now.